zebra-chain/src/value_balance.rs

```diff
@@ -184,3 +184,28 @@ where
         self? - rhs
     }
 }
+
+impl<C> std::iter::Sum<ValueBalance<C>> for Result<ValueBalance<C>, ValueBalanceError>
+where
+    C: Constraint + Copy,
+{
+    fn sum<I: Iterator<Item = ValueBalance<C>>>(iter: I) -> Self {
+        iter.fold(Ok(ValueBalance::zero()), |acc, value_balance| {
+            Ok(ValueBalance {
+                transparent: (acc.clone()?.transparent + value_balance.transparent)?,
```

Comment:
We should try to avoid this clones.

Reply:
Can you use `try_fold::Result<ValueBalance, ValueBalanceError>` here?

The code should be a lot simpler than `impl<C> std::iter::Sum<Amount<C>> for Result<Amount<C>>`, because the value and error types are exactly what you need.
https://github.com/ZcashFoundation/zebra/blob/a7b0a521d51af4af7294892604c34087e0414330/zebra-chain/src/amount.rs#L304

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.try_fold